libpod/image/image.go

```diff
@@ -1075,3 +1075,59 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+//LayerInfo keeps information of single layer
+//along with ID of its parent and child layers.
+type LayerInfo struct {
+	ID       string
+	ParentID string
+	//There can be multiple childs (fork)
+	ChildID  []string
+	RepoTags []string
+	Size     int64
+}
+
+// GetLayersMapWithImageInfo returns map of image-layers, with associated information like RepoTags, parent and list of child layers.
+func (i *Image) GetLayersMapWithImageInfo(ctx context.Context) (map[string]*LayerInfo, error) {
+	imgs, _ := i.imageruntime.store.Images()
+	layers, _ := i.imageruntime.store.Layers()
+
+	// Builds the layer dependency chain with ParentID and ChildID
+	layerInfoMap := make(map[string]*LayerInfo)
+
+	// scan all layers & fill size and parent id
+	for _, layer := range layers {
+		_, ok := layerInfoMap[layer.ID]
+		if !ok {
+			layerInfoMap[layer.ID] = &LayerInfo{
+				ID:       layer.ID,
+				Size:     layer.UncompressedSize,
+				ParentID: layer.Parent,
+			}
+		} else {
+			return nil, fmt.Errorf("data inconsistency : %s layer-id have multiple entry in layer store", layer.ID)
+		}
+	}
+
+	// scan all layers & add all childs for each layers to layerInfo
+	for _, layer := range layers {
+		_, ok := layerInfoMap[layer.ID]
+		if ok {
+			if layer.Parent != "" {
+				layerInfoMap[layer.Parent].ChildID = append(layerInfoMap[layer.Parent].ChildID, layer.ID)
+			}
+		} else {
+			return nil, fmt.Errorf("lookup error: layer-id  %s, not found", layer.ID)
+		}
+	}
+
+	// Add the Repo Tags to Top layer of each image.
+	for _, img := range imgs {
```

Comment:
(It would be a tiny bit more readable to inline the value of `imgs` here and avoid the variable, or at the very least to compute `imgs` right here and not all the way at the top of the function.)